spinaltap-common/src/main/java/com/airbnb/spinaltap/common/destination/DestinationPool.java

```diff
@@ -71,38 +78,26 @@ public int getPoolSize() {
   }
 
   /**
-   * Partitions the mutation list according to the supplied key provider and routes to the
-   * corresponding destinations
+   * Partitions the {@link Mutation} list according to the supplied {@link KeyProvider} and routes
+   * to the corresponding {@link Destination}s
    */
   @Override
-  public synchronized void send(List<? extends Mutation<?>> mutations) {
-    partition(mutations)
+  public synchronized void send(@NonNull final List<? extends Mutation<?>> mutations) {
+    mutations
+        .stream()
+        .collect(groupingBy(this::getPartitionId))
```

Comment:
This won't retain `partitionId` order in `mutations` as the old code used a `LinkedHashMap`. Don't we need to retain this order? @jadabisamra